### ISSUE TITLE
Being caught carrying illegal cargo for a mission should trigger the "on fail" tag and fail the mission

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#556]
Otherwise, it seems kinda weird. They catch you carrying illegal cargo, fine you, but let you keep the contraband? What government has ever done that?

Also, making missions fail if you get scanned opens the door to missions where you have to avoid getting scanned or else you fail.

**Comment by thenerdfreak** [comment]
Kinda sucks tho, then you'd lose money on the fine and not even get to finish your mission. And Republic ships are way too scanhappy

**Comment by Pointedstick** [comment]
Not anymore with the latest changes. I'm writing a mission that you're supposed to fail if you get scanned and I had a hack of a time getting Republic ships to actually scan me!

**Comment by Nightmareman** [comment]
They are still way to scan happy for me some systems I still get scanned 30 times even though that was supposed to be reduced. Also even if they do scan you and see the illegal cargo how are they going to get it all they could do is order you to let them board or drop it which you could just ignore or kill them for scanning you which is generally what I do.

**Comment by bobafetthotmail** [comment]
Haha, everyone is right.

Nightmareman is right, in case they find illegal cargo you should be offered the choice to drop illegal cargo or not, if yes you just get fined (and lose the object), if not they fine you and attack you.

But Pointedstick is also right.
There can be missions where not getting caught at all might make sense as a mission objective.
Most secret/covert missions need that. 
I mean, if the enemy knows that someone is carrying a particular top secret object around without permission, they will probably trace back your movements and track down who gave you the mission pretty easily, or just follow you and see who gets the delivery.
Or just assume that their enemies got that top secret object and react accordingly.

But thenerdfreak is also right.
Getting scanned is random, you can get scanned 30 times or not scanned any time.
The main outcome of a scan with such trigger would be to just have me reload a save, and try again.
That would be just annoying. There needs to be some way to elude scans that involves player skill.
Like avoiding the scan ships, or a minigame where you jam their scanners (like the minigames to open locks in Mass Effect or most half-decent RPGs like say Gothic series).

**Comment by ph2000bis** [comment]
Mini-games ? No thanks. However, flying skill, maybe by reducing the scanning radius, would involve evading the scan ships

**Comment by bobafetthotmail** [comment]
Scan ships are usually the light warships, fast ships. I doubt it will be enough.

I was thinking about making the scanner not instantaneous.
Like say you need to stay inside a short range for 20 seconds or whatever (continuous or not) for it to work.

This way you can get notified that someone is STARTING to scan you and you can shout "oh crap!", and try to evade him before the time is up and the scan is complete.

would also be nice to give the scanner a obvious weapon-like effect in-game so I can see who is scanning me, so I know who I need to evade.

**Comment by Pointedstick** [comment]
> There needs to be some way to elude scans that involves player skill.

Yes! This is a great idea. It would also provide an incentive to have a light and speedy ship, which would probably be able to evade scans much more easily than a huge and cumbersome capital ship.

**Comment by Pointedstick** [comment]
I encourage you to break this out into a new issue, @bobafetthotmail. I think it's a great idea.

**Comment by ph2000bis** [comment]
What do you mean weapons that prevent you from jumping?

**Comment by thenerdfreak** [comment]
weapons that do knockback and prevent you from aligning for jumps (the particle cannon is a good example)

**Comment by frameworld** [comment]
Scanner ships are also surveillance drones on carriers.
They don't even bother equipping hyper-drives or weapons on these.

@Nightmareman each government scans you only once a day. Days don't pass while flying through a system, only while jumping or landing.
That means in a system with 30 governments, you get scanned 30x.

With a fleet of 30 ships, do you get scanned 30x? With 2 governments and 15 ships you get scanned 30x too?

**Comment by Nightmareman** [comment]
@frameworld Incorrect as I have had only 1 ship in my fleet and still get scanned 30 times by the republic. It was supposed to be only once a day but it doesn't actually function like that. Main source of the problem is ships with the surveillance personality will scan you regardless of you having been scanned before. So if you end up in a system where multiple ships have surveillance personality the bottom of the screen can light up with multiple scans all at the same time. Which is why anymore when ships start scanning me I start shooting them I have good enough reputation from all the pirates I have killed to suffer the small hit of evaporating the ships that try and scan me.

**Comment by frameworld** [comment]
You did not say you had just 1 ship.
I can only guess that if all scans start in the same split-second, they can't notify each other of it being successful.

A quick solution would be if they add the scan-markers/attachments on your ship similar to a pin-note.
In the next frame/split-second a loop takes all scan-markers and the first of each government completes a successful scan - all others fail.
 * That even works with multi-threading if the pin-note script works correctly with it.

**Comment by Pointedstick** [comment]
This came up again for a mission I'm writing. I'd like it to fail if you're scanned. Right now there's no way to do this.

**Comment by Pointedstick** [comment]
Mostly addressed with https://github.com/endless-sky/endless-sky/pull/1584.

**Comment by MarcelineVQ** [comment]
The outstanding scanning bug is from a ship with both cargo and outfit scanners trying to scan you. Cargo and outfit scanning have different ranges and a ship with both will repeatedly scan with whatever is in range until both have been done. Typically this results in instant fines for Nerve Gas since outfit scanning is in range before cargo scanning so outfits get scanned many times.

**Comment by Pointedstick** [comment]
That's true, but this  issue is solely about mission cargo, not outfits. Separate issue.

**Comment by MarcelineVQ** [comment]
Yep, just informing people about why they might be seeing inflated scan occurrences.

**Comment by Pointedstick** [comment]
Between https://github.com/endless-sky/endless-sky/pull/1584 and https://github.com/endless-sky/endless-sky/commit/40d87c474064e7ba5ac8d7f5bd902a445095bb8e, this is resolved now!

**Comment by DingusShingleton** [comment]
I had a serious mental blue screen looking at the dates for this thread..... "Jan 1??? That's in a week! 2016!?!? What's going on!?  Oh wait. The upcoming year is 20*17*."

I am not a clever man.

**Comment by KiLEdEnNis** [comment]
@DingusShingleton that's why men marry women that can do everything we can't
(side note; not meant as racism of any kind)

**Comment by Amazinite** [comment]
> racism

**Comment by KiLEdEnNis** [comment]
;-;

**Comment by Pointedstick** [comment]
![304](https://cloud.githubusercontent.com/assets/1097249/21463354/5fbe3daa-c925-11e6-9a0c-32bcd56238d6.gif)

**Comment by Amazinite** [comment]
Lmfao. I saw that the notification had no text, and I immediately thought that you had posted that. I was too right. XD

**Comment by KiLEdEnNis** [comment]
( ͡°, _ʖ ͡°,)

**Comment by jafdy** [comment]
This is not the place for off topic.
This is -> https://discord.gg/u6KepgG
